### PR TITLE
initial commit for subtheme drupal console command

### PR DESCRIPTION
**Pull request by Chris7791**

### CODE REVIEW

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+ *     extension="origins_subtheme",
+ *     extensionType="module"
+ * )
+ */
+class OriginsSubthemeCommand extends ContainerAwareCommand {
+
+  /**
+   * @var Io.
+   */
+  private $Io;
```
Comment (DuttonMa) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: I like the fact that you have thought about this and used 'private'

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        'Select which Theme to Subtheme off.',
+        array_keys($this->parent_themes)
+      );
+    }
+    $input->setOption('parent_theme', $parent_theme); // Machine name set as value.
+
+    // Set Description (OPTIONAL)
+    $description = $this->Io->askEmpty(
+      'Please provide a description for the subtheme',
+      'Subtheme of ' . $parent_theme
```
Comment (DuttonMa) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: Not a big deal, but would generally expect to see placeholders used here

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+  }
+
+  /**
+   * Cleans up name to be machine friendly.
+   *
+   * @param $name
+   * @return mixed
+   */
+  public function machine_name_validation($name)
+  {
```
Comment (DuttonMa) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: Nice function

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+   *
+   * @return array
+   */
+  public function get_startkit_themes()
+  {
+    // Init Var.
+    $parentThemeOptions = [];
+
+    // Loads all Installed Themes.
+    $themes = \Drupal::service('theme_handler')->listInfo();
```
Comment (DuttonMa) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: You should use dependency injection here, as a general rule Drupal services shouldn't be called directly from a class method

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+  }
+
+  /**
+   * Iterates Directories and renames files as well as file contents.
+   *
+   * @param $parent_theme
+   * @param $subtheme_name
+   */
+  public function iterate_directory($subtheme_name)
+  {
```
Comment (DuttonMa) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: This is a nice well written function !

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    $file_content_search = new Finder();
+
+    // Finds all files with Placeholder in file name.
+    $file_name_search->name('*'.self::PLACEHOLDER.'*')->in(self::BASE_THEME_DIRECTORY . $subtheme_name);
+
+    foreach ($file_name_search as $result) {
+      //foreach file rename using PHP rename function.
+      rename(
+        self::BASE_THEME_DIRECTORY . $subtheme_name . '/' . $result->getFilename(),
+        self::BASE_THEME_DIRECTORY . $subtheme_name .'/'. str_replace(self::PLACEHOLDER, $subtheme_name, $result->getFilename())
```
Comment (johangant) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: String concatenators need a space either side.

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+      );
+    }
+
+    // Finds all file with placeholder in file contents.
+    $file_content_search->contains('/STARTERKIT/')->in(self::BASE_THEME_DIRECTORY . $subtheme_name);
+
+    foreach ($file_content_search as $result) {
+      // info yml special case
```
Comment (johangant) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]:
```diff
- // info yml special case
+ // Info yml special case.
```

Comment style needs to be phpcs compliant.

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+      if(strpos($result->getFilename(), 'info.yml')) {
```
Comment (johangant) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: Need a space between if and the opening bracket.

```diff
- if(strpos....
+ if (strpos....
```

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+          $info_yml_array = Yaml::parseFile(self::BASE_THEME_DIRECTORY . $subtheme_name . '/' . $result->getFilename());
+          // Set Name attribute to User Input name
+          $info_yml_array['name'] = $this->friendly_name;
+          // Set Description
+          $info_yml_array['description'] = $this->description;
+          // Dump array into YAML format
+          $yaml = Yaml::dump($info_yml_array);
+          // Set new contents in YAML file
+          file_put_contents(self::BASE_THEME_DIRECTORY . $subtheme_name . '/' . $result->getFilename(), $yaml);
+        } catch (ParseException $exception) {
```
Comment (johangant) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: Catch on newline, please

**Comment by DuttonMa** [comment]
I see that you have quite a lot of coding standards failures, you may want to run your code through phpcs before creating the PR

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+          $this->Io->info('Unable to parse the info YAML file: ' . $exception->getMessage());
+        }
+      }
+
+      // Load each files contents
+      $contents = file_get_contents(self::BASE_THEME_DIRECTORY . $subtheme_name .'/' . $result->getFilename());
```
Comment (johangant) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: String concatenators with a single space either side, please

File: origins_subtheme/src/Command/OriginsSubthemeCommand.php [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+      file_get_contents(self::BASE_THEME_DIRECTORY . $subtheme_name . '/' . $result->getFilename());
```
Comment (omahm) [inline origins_subtheme/src/Command/OriginsSubthemeCommand.php]: What is the purpose of this line?

**Pull request closed without merging**